web/toast_manager.js

```diff
@@ -0,0 +1,77 @@
+/* Copyright 2024 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class ToastManager {
```

Comment:
I'm not convinced this is a great name, which applies everywhere through the PR, since this only handles AnnotationEditor undo and nothing else really.

Maybe `EditorUndoBar` instead, since that seems more appropriate?